### ISSUE TITLE
dyld[88714]: symbol not found in flat namespace '_libintl_bindtextdomain'

### ISSUE THREAD

**Issue by kevalshah90** [pyenv/pyenv#2611]
```
pyenv install 3.10
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-3.10.9.tar.xz...
-> https://www.python.org/ftp/python/3.10.9/Python-3.10.9.tar.xz
Installing Python-3.10.9...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 13.0.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/d0/gnksqzwn2fn46fjgrkp6045c0000gn/T/python-build.20230201224331.77801
Results logged to /var/folders/d0/gnksqzwn2fn46fjgrkp6045c0000gn/T/python-build.20230201224331.77801.log

Last 10 log lines:
DYLD_LIBRARY_PATH=/var/folders/d0/gnksqzwn2fn46fjgrkp6045c0000gn/T/python-build.20230201224331.77801/Python-3.10.9 ./python.exe -E -S -m sysconfig --generate-posix-vars ;\
	if test $? -ne 0 ; then \
		echo "generate-posix-vars failed" ; \
		rm -f ./pybuilddir.txt ; \
		exit 1 ; \
	fi
dyld[88714]: symbol not found in flat namespace '_libintl_bindtextdomain'
/bin/sh: line 1: 88714 Abort trap: 6           DYLD_LIBRARY_PATH=/var/folders/d0/gnksqzwn2fn46fjgrkp6045c0000gn/T/python-build.20230201224331.77801/Python-3.10.9 ./python.exe -E -S -m sysconfig --generate-posix-vars
generate-posix-vars failed
make: *** [pybuilddir.txt] Error 1
```

### Description
- [ MacOS Ventura 13.0.1] Platform information (e.g. Ubuntu Linux 16.04):
- [ ] OS architecture (e.g. Apple M1 Chip):
- [2.13.2 ] pyenv version:

**Comment by native-api** [comment]
Please add the required diagnostic infgormation as per the issue template.

**Comment by minseung-pf** [comment]
This worked for me:
`arch --arm64 pyenv install 3.10`

**Comment by cyber237** [comment]
Facing same problem

**Comment by cyber237** [comment]
Just found a fix
@kevalshah90 use gcc-11

$ brew install gcc@11 && CC=gcc-11 pyenv install python 3.10

**Comment by CarpathianUA** [comment]
@kevalshah90 
It can be related to inconsistency between versions of libraries you have locally - e.g. if you migrated from Intel-based mac to M1/M2. I had exactly the same error while installing Python with `pyenv install`. I went over `/usr/local/Cellar` `/usr/local/bin` and similar folders and cleaned up all symlinks and binaries of `xz`, `gettext`, `gcc` and `readline`, then installed them from scratch with Homebrew (they will appear under `/opt/homebrew`, you can also check their path with `brew info <package_name>`)

**Comment by thomasnal** [comment]
```sh
$ pyenv install 3.10.9
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Installing Python-3.10.9...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew

BUILD FAILED (OS X 12.3 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/47/kr3ddq2j1wx3t1223dtn277c0000gn/T/python-build.20230212161344.59085
Results logged to /var/folders/47/kr3ddq2j1wx3t1223dtn277c0000gn/T/python-build.20230212161344.59085.log

Last 10 log lines:
DYLD_LIBRARY_PATH=/var/folders/47/kr3ddq2j1wx3t1223dtn277c0000gn/T/python-build.20230212161344.59085/Python-3.10.9 ./python.exe -E -S -m sysconfig --generate-posix-vars ;\
	if test $? -ne 0 ; then \
		echo "generate-posix-vars failed" ; \
		rm -f ./pybuilddir.txt ; \
		exit 1 ; \
	fi
dyld[69595]: symbol not found in flat namespace '_libintl_bindtextdomain'
/bin/sh: line 1: 69595 Abort trap: 6           DYLD_LIBRARY_PATH=/var/folders/47/kr3ddq2j1wx3t1223dtn277c0000gn/T/python-build.20230212161344.59085/Python-3.10.9 ./python.exe -E -S -m sysconfig --generate-posix-vars
generate-posix-vars failed
make: *** [pybuilddir.txt] Error 1
```

[python-build.20230212161932.80571.log](https://github.com/pyenv/pyenv/files/10716374/python-build.20230212161932.80571.log)

**Comment by native-api** [comment]
@thomasnal Not enough information provided, see the issue template.
From what is provided, I can only see that 1) you're building for ARM64 but using x64 Homebrew 2) gettext is installed but not usable -- probably installed for the wrong architecture, same as in https://github.com/pyenv/pyenv/issues/2573#issuecomment-1415335052.

**Comment by thomasnal** [comment]
```sh
$ pyenv --version
pyenv 2.3.12
$ python -V
Python 3.7.4
$ pyenv versions
  system
* 3.7.4 (set by /Users/thomas/.pyenv/version)
  3.8.3
  3.9.7
  3.10.0
$ gcc -v
Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
$ env PYENV_DEBUG=1 pyenv install -v 3.10.9 2>&1 | tee trace.log
[trace.log attached below]

```

[trace.log](https://github.com/pyenv/pyenv/files/10716621/trace.log)

[config.log](https://github.com/pyenv/pyenv/files/10716627/config.log)

**Comment by native-api** [comment]
@thomasnal Yes, you have `gettext` installed in the x64 Homebrew, same problem as in https://github.com/pyenv/pyenv/issues/2573#issuecomment-1415335052. Uninstall it there or install it in the Arm64 Homebrew.
```
configure:10344: checking for textdomain in -lintl
configure:10369: clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include  -I/usr/local/opt/readline/include -I/usr/local/opt/readline/include -I/Users/thomas/.pyenv/versions/3.10.9/include -I/usr/local/opt/openssl@1.1/include -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/Users/thomas/.pyenv/versions/3.10.9/lib -Wl,-rpath,/Users/thomas/.pyenv/versions/3.10.9/lib -L/usr/local/opt/openssl@1.1/lib -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib conftest.c -lintl  -ldl  >&5
ld: warning: directory not found for option '-L/Users/thomas/.pyenv/versions/3.10.9/lib'
ld: warning: ignoring file /usr/local/lib/libintl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_textdomain", referenced from:
      _main in conftest-a97c62.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by thomasnal** [comment]
Homebrew installed in earlier days of apple silicon used Rosetta, installed and ran x86 binaries. These days, homebrew install sets up on arm64 architecture.

Force uninstalling gettext didn't help, another error appeared. I have reinstalled homebrew that allowed pyenv to find arm64 version of gettext and it compiles the python.

Note: Now I wonder how to install x86 version of gettext alongside of arm version. All installed python versions and environments are not able to run their apps, all look for x86 version of gettext. Most dev tools like neovim are broken too, despite reinstalled via homebrew.  Much more headache and effort is needed.

```sh
$ brew bundle dump
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"
$ /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" 
$ brew bundle install
$ pyenv install 3.10.9
python-build: use openssl@1.1 from homebrew
python-build: use readline from homebrew
Installing Python-3.10.9...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk
Installed Python-3.10.9 to /Users/thomas/.pyenv/versions/3.10.9
```

**Comment by native-api** [comment]
We really need guidance on how to use Pyenv on Apple Silicon in the wiki. I can't write it since I don't have the hardware to verify the instructions, someone else has to.

https://github.com/pyenv/pyenv/issues/1853 has the majority of the guidance. It references https://stackoverflow.com/questions/64951024/how-can-i-run-two-isolated-installations-of-homebrew which suggests to create two shortcuts for console -- for arm64 and for x64, each with its own Homebrew in path.

Other than install Pyenv 2 times for each Homebrew as suggested there, you can install it once via Git checkout. You're still better set different `PYENV_ROOT`s for each architecture.

**Comment by nayt140** [comment]
> @thomasnal Yes, you have `gettext` installed in the x64 Homebrew, same problem as in [#2573 (comment)](https://github.com/pyenv/pyenv/issues/2573#issuecomment-1415335052). Uninstall it there or install it in the Arm64 Homebrew. Or build for x64.
> 
> ```
> configure:10344: checking for textdomain in -lintl
> configure:10369: clang -o conftest -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include  -I/usr/local/opt/readline/include -I/usr/local/opt/readline/include -I/Users/thomas/.pyenv/versions/3.10.9/include -I/usr/local/opt/openssl@1.1/include -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/Users/thomas/.pyenv/versions/3.10.9/lib -Wl,-rpath,/Users/thomas/.pyenv/versions/3.10.9/lib -L/usr/local/opt/openssl@1.1/lib -L/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib conftest.c -lintl  -ldl  >&5
> ld: warning: directory not found for option '-L/Users/thomas/.pyenv/versions/3.10.9/lib'
> ld: warning: ignoring file /usr/local/lib/libintl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
> Undefined symbols for architecture arm64:
>   "_textdomain", referenced from:
>       _main in conftest-a97c62.o
> ld: symbol(s) not found for architecture arm64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> ```

Yep installing gettext on arm version of brew did the trick.

**Comment by ut-adamc** [comment]
> Just found a fix @kevalshah90 use gcc-11
> 
> $ brew install gcc@11 && CC=gcc-11 pyenv install python 3.10

I tried this, but I ended up with the same error as. before:
`DYLD_LIBRARY_PATH=/var/folders/wc/01hqy4890zv7n7wnqnkjr7vc0000gq/T/python-build.20230309154327.76248/Python-3.8.16 ./python.exe -E -S -m sysconfig --generate-posix-vars ;\
	if test $? -ne 0 ; then \
		echo "generate-posix-vars failed" ; \
		rm -f ./pybuilddir.txt ; \
		exit 1 ; \
	fi
dyld[91317]: symbol not found in flat namespace (_libintl_bindtextdomain)
/bin/sh: line 1: 91317 Abort trap: 6           DYLD_LIBRARY_PATH=/var/folders/wc/01hqy4890zv7n7wnqnkjr7vc0000gq/T/python-build.20230309154327.76248/Python-3.8.16 ./python.exe -E -S -m sysconfig --generate-posix-vars
generate-posix-vars failed
make: *** [pybuilddir.txt] Error 1`

**Comment by smancebo** [comment]
> @kevalshah90 It can be related to inconsistency between versions of libraries you have locally - e.g. if you migrated from Intel-based mac to M1/M2. I had exactly the same error while installing Python with `pyenv install`. I went over `/usr/local/Cellar` `/usr/local/bin` and similar folders and cleaned up all symlinks and binaries of `xz`, `gettext`, `gcc` and `readline`, then installed them from scratch with Homebrew (they will appear under `/opt/homebrew`, you can also check their path with `brew info <package_name>`)

Same happened to me as well, had to re-install brew (to use the arm64 version) and remove those libs from /usr/local/Cellar, after that i was able to install the python version with pyenv

**Comment by promlife** [comment]
![image](https://user-images.githubusercontent.com/3366962/227418608-caf9eed8-59df-492e-bf93-51fe7d23c2c5.png)

I have the same problem recently when in use 

env:

- MAC OS 13
- m1 max
- pyenv 2.3.15
- Xcode 14.1(Build version 14B47b)

**Comment by promlife** [comment]
> ![image](https://user-images.githubusercontent.com/3366962/227418608-caf9eed8-59df-492e-bf93-51fe7d23c2c5.png)
> 
> I have the same problem recently when in use
> 
> env:
> 
> * MAC OS 13
> * m1 max
> * pyenv 2.3.15
> * Xcode 14.1(Build version 14B47b)

I works for me by using  "arch -arm64 pyenv install 3.9.11"

**Comment by MaximMukhametov** [comment]
> readline

Works for me as well

pyenv 2.3.16
macOS 13.2.1 (22D68) 
Apple M1 Pro

**Comment by MaximMukhametov** [comment]
> @kevalshah90 It can be related to inconsistency between versions of libraries you have locally - e.g. if you migrated from Intel-based mac to M1/M2. I had exactly the same error while installing Python with `pyenv install`. I went over `/usr/local/Cellar` `/usr/local/bin` and similar folders and cleaned up all symlinks and binaries of `xz`, `gettext`, `gcc` and `readline`, then installed them from scratch with Homebrew (they will appear under `/opt/homebrew`, you can also check their path with `brew info <package_name>`)

Works for me as well

pyenv 2.3.16
macOS 13.2.1 (22D68)
Apple M1 Pro

**Comment by tsuz** [comment]
None of the above worked for me but this worked:

```
rm -rf /usr/local/Cellar/gettext
rm -rf /usr/local/Cellar/openssl
rm -rf /usr/local/Cellar/readline
rm -rf /usr/local/Cellar/tcl-tk
export PYTHON_BUILD_HOMEBREW_OPENSSL_FORMULA=openssl@3.0
pyenv install 3.10.3  
```

**Comment by Kyuubi1** [comment]
> @kevalshah90 It can be related to inconsistency between versions of libraries you have locally - e.g. if you migrated from Intel-based mac to M1/M2. I had exactly the same error while installing Python with `pyenv install`. I went over `/usr/local/Cellar` `/usr/local/bin` and similar folders and cleaned up all symlinks and binaries of `xz`, `gettext`, `gcc` and `readline`, then installed them from scratch with Homebrew (they will appear under `/opt/homebrew`, you can also check their path with `brew info <package_name>`)

U save my life with this

**Comment by petergaultney** [comment]
I was running into this when trying to add an arm64 installation of pyenv next to my x86_64 Rosetta installation.

I tried various things suggested here, none of which worked.

I did notice that when running `pyenv install 3.8.17 -vvv`, the verbose logging messages showed that it was still trying to build x86_64.

I ended up manuallly removing `/usr/bin/local` (where my x86_64 `brew` install lives) from `$PATH`. Having done this, I now get `checking build system type... arm-apple-darwin21.6.0` during the build, which succeeds.

I still  can't figure out how that's getting on my PATH in the first place, nor why it somehow takes precedence over ARM `brew`. Probably some underlying subshell is getting created somewhere.
